schema/me/update_me_mutation.js

```diff
@@ -0,0 +1,81 @@
+import { GraphQLString, GraphQLInt, GraphQLInputObjectType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+
+import UserType from "../user"
+
+export const EditableLocationFields = new GraphQLInputObjectType({
+  name: "EditableLocation",
+  fields: {
+    address: {
+      description: "First line of an address",
+      type: GraphQLString,
+    },
+    address_2: {
+      description: "Second line of an address",
+      type: GraphQLString,
+    },
+    city: {
+      description: "The city the location is based in",
+      type: GraphQLString,
+    },
+    country: {
+      description: "The county the location is based in",
+      type: GraphQLString,
+    },
+    summary: {
+      description: "An optional display string for the location",
+      type: GraphQLString,
+    },
+    postal_code: {
+      description: "Postal code for a string",
+      type: GraphQLString,
+    },
+    state: {
+      description: "The (optional) name of the state for location",
+      type: GraphQLString,
+    },
+    state_code: {
+      description: "The (optional) state code of the state for location",
+      type: GraphQLString,
+    },
+  },
+})
+
+export default mutationWithClientMutationId({
+  name: "UpdateMyProfile",
+  description: "Update the current logged in user.",
+  // This is just a subset needed today, anything inside the User object
+  // in gravity can go in here as needed.
+  inputFields: {
+    name: {
+      description: "The given name of the user.",
+      type: GraphQLString,
+    },
+    email: {
+      description: "The given email of the user.",
+      type: GraphQLString,
+    },
+    phone: {
+      description: "The given phone number of the user.",
+      type: GraphQLString,
+    },
+    location: {
+      description: "The given location of the user as structured data",
+      type: EditableLocationFields,
+    },
+    collector_level: {
+      description: "The collector level for the user",
+      type: GraphQLInt,
+    },
+  },
+  outputFields: {
+    user: {
+      type: UserType,
+      resolve: user => user,
+    },
+  },
+  mutateAndGetPayload: (user, _request, { rootValue: { updateMeLoader } }) => {
+    if (!updateMeLoader) throw new Error("No updateMeLoader loader found in root values")
```

Comment:
I'm down with raising an error (instead of `return null`). What are the reasons this might not exist, if the required auth headers are missing or something? Or is just a guard in case someone modifies that other file not realizing it was being used?

Reply:
Honestly, I don't know, I'm sure at some point I saw a comment around it as being testing purposes. But, if you make a mutation and return null, relay crashes - [you literally have to return something](https://stackoverflow.com/questions/33831911/cannot-set-property-clientmutationid-of-undefined)

So, I think it's a _~bad_ protection against dev error, which I think a `throw` does a better job of expressing

Reply:
Makes sense 👍

Reply:
Authenticated loaders won’t exist if there was no access token header specified.